src/seat/keyboard/mod.rs

```diff
@@ -568,10 +589,23 @@ where
 
                         match state {
                             wl_keyboard::KeyState::Released => {
+                                #[cfg(feature = "calloop")]
+                                {
+                                    if let Some(repeat_sender) = &udata.repeat_sender {
+                                        let _ = repeat_sender.send(RepeatMessage::StopRepeat);
+                                    }
+                                }
                                 data.release_key(conn, qh, keyboard, serial, event);
                             }
 
                             wl_keyboard::KeyState::Pressed => {
+                                #[cfg(feature = "calloop")]
+                                {
+                                    if let Some(repeat_sender) = &udata.repeat_sender {
```

Comment:
xkbcommon has keys which should not be repeated. I believe there is a way to check for those keys on the keymap?